compiler/rustc_codegen_ssa/src/mir/block.rs

```diff
@@ -376,20 +376,29 @@ impl<'a, 'tcx, Bx: BuilderMethods<'a, 'tcx>> FunctionCx<'a, 'tcx, Bx> {
             // If there are two targets (one conditional, one fallback), emit `br` instead of
             // `switch`.
             let (test_value, target) = target_iter.next().unwrap();
+            let otherwise = targets.otherwise();
+            let true_cold = self.cold_blocks[target];
+            let false_cold = self.cold_blocks[otherwise];
+            let expect = if true_cold != false_cold { Some(false_cold) } else { None };
```

Comment:
I had to scratch my head for quite some time to understand why you wrote it this way. IIUC:
- if `true_cold == false_cold`, both branches have the same weight, so we don't emit anything;
- if `false_cold` and `1 => target` we expect to go to `lltrue`, so we `expect` is `Some(true)`;
- ...

This definitely needs some comment.

Reply:
Commend added